src/Kros.MediatR.Extensions/ServiceCollectionExtensions.cs

```diff
@@ -63,18 +63,38 @@ public static IServiceCollection AddPipelineBehaviorsForRequest<TRequest>(this I
             return services;
         }
 
-        private static IEnumerable<Type> GetPipelineBehaviors(Type requestType, Type pipeLineType, Assembly behavioursAssembly)
-            => (behavioursAssembly ?? Assembly.GetAssembly(requestType))
+        private static IEnumerable<Type> GetPipelineBehaviors(
+            Type requestType,
+            Type pipeLineType,
+            IEnumerable<Assembly> behaviorsAssemblies)
+        {
+            if (behaviorsAssemblies is null || !behaviorsAssemblies.Any())
+            {
+                return GetPipelineBehaviors(requestType, pipeLineType, Assembly.GetAssembly(requestType)).ToList();
+            }
+            return behaviorsAssemblies.SelectMany(a => GetPipelineBehaviors(requestType, pipeLineType, a));
+        }
+
+        private static IEnumerable<Type> GetPipelineBehaviors(Type requestType, Type pipeLineType, Assembly behaviorsAssembly)
+            => behaviorsAssembly
             .GetTypes()
             .Where(t
                 => t.GetInterface(pipeLineType.Name) != null
                 && t.GetGenericArguments()[0].GetInterface(requestType.Name) != null);
 
-        private static IList<Type> GetTypes(Type type, Assembly assembly)
-            => (assembly ?? Assembly.GetAssembly(type))
+        private static IList<Type> GetTypes(Type type, IEnumerable<Assembly> assemblies)
```

Comment:
Stačí keď to bude `IEnumerable` (aj premenná na riadku 35). A zrušiť volania `ToList()`.

Reply:
na tom riadku 35 som nechal list. Tato kolekcia tam ide do vnoreneho foreach a scannovali by sa tam assebly dookola